File: 350. Intersection of Two Arrays II.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        res=[]
        for n1 in nums1:
            if n1 in nums2:
                nums2.remove(n1)
                res += [n1]
                
        return res

File: 350. Intersection of Two Arrays II.cpp

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        map<int,int> m;
        vector<int> res;
        
        int i;
        for(i=0;i<nums1.size();i++){
            m[nums1[i]]++;
        }
        
        for(i=0;i<nums2.size();i++){
            if(m[nums2[i]]>=1){
                m[nums2[i]]--;
                res.push_back(nums2[i]);
            }
        }
        return res;
        
    }
};
